google-cloud-core/src/test/java/com/google/cloud/TimestampTest.java

```diff
@@ -104,6 +112,14 @@ public void ofSqlTimestampOnEpoch() {
     assertThat(timestamp.toString()).isEqualTo(expectedTimestampString);
   }
 
+  @Test
+  public void testOf_preEpochExactSecond() {
+    String expectedTimestampString = "1969-12-31T23:59:59Z";
+    java.sql.Timestamp input = new java.sql.Timestamp(-1000);
+    Timestamp timestamp = Timestamp.of(input);
+    assertThat(timestamp.toString()).isEqualTo(expectedTimestampString);
```

Comment:
gratuitous use of assertThat; assertEquals is simpler here

Reply:
Done.